src/lib/operators/projection.cpp

```diff
@@ -120,27 +120,39 @@ std::shared_ptr<const Table> Projection::_on_execute() {
-                    const auto& dictionary = typed_segment.dictionary();
-
-                    auto filtered_attribute_vector = pmr_vector<ValueID::base_type>(pos_list->size());
-
-                    auto iterable = create_iterable_from_attribute_vector(typed_segment);
+                  // Write new attribute vector containing the positions given from the input_pos_list and compress it
+                  [[maybe_unused]] auto create_filtered_and_compressed_pos_list = [](const auto& input_segment,
+                                                                                     const auto& input_pos_list) {
+                    auto filtered_attribute_vector = pmr_vector<ValueID::base_type>(input_pos_list->size());
+                    auto iterable = create_iterable_from_attribute_vector(input_segment);
                     auto chunk_offset = ChunkOffset{0};
-                    iterable.with_iterators(pos_list, [&](auto it, auto end) {
+                    iterable.with_iterators(input_pos_list, [&](auto it, auto end) {
                       while (it != end) {
                         filtered_attribute_vector[chunk_offset] = it->value();
                         ++it;
                         ++chunk_offset;
                       }
                     });
+                    return std::make_shared<FixedSizeByteAlignedVector<uint32_t>>(std::move(filtered_attribute_vector));
+                  };
+
+                  if constexpr (std::is_same_v<DictionarySegmentType, DictionarySegment<ColumnDataType>>) {
+                    const auto compressed_attribute_vector =
+                        create_filtered_and_compressed_pos_list(typed_segment, pos_list);
+                    const auto& dictionary = typed_segment.dictionary();
 
-                    auto compressed_attribute_vector =
-                        compress_vector(filtered_attribute_vector, VectorCompressionType::FixedSizeByteAligned, {});
                     output_segments[column_id] = std::make_shared<DictionarySegment<ColumnDataType>>(
                         dictionary, std::move(compressed_attribute_vector),
                         referenced_dictionary_segment->null_value_id());
+                  } else if constexpr (std::is_same_v<DictionarySegmentType,  // NOLINT
+                                                      FixedStringDictionarySegment<ColumnDataType>>) {
+                    const auto compressed_attribute_vector =
+                        create_filtered_and_compressed_pos_list(typed_segment, pos_list);
+                    const auto& dictionary = typed_segment.fixed_string_dictionary();
+
+                    output_segments[column_id] = std::make_shared<FixedStringDictionarySegment<ColumnDataType>>(
+                        dictionary, std::move(compressed_attribute_vector),
+                        referenced_dictionary_segment->null_value_id());
```

Comment:
Hacked together a quick test for this (sans FixedString). Could you add it?

```
TEST_F(OperatorsProjectionTest, ReusesDictionaryWhenForwarding) {
  // Checks that instead of materializing all values in an imperfectly forwarded DictionarySegment, the dictionary
  // is re-used and only the attribute vector is materialized.

  auto dict_table = load_table("resources/test_data/tbl/int_float4.tbl", 3);
  ChunkEncoder::encode_all_chunks(dict_table, EncodingType::Dictionary);
  auto table_wrapper = std::make_shared<TableWrapper>(dict_table);
  table_wrapper->execute();

  const auto table_scan = create_table_scan(table_wrapper, ColumnID{0}, PredicateCondition::Equals, 123456);
  table_scan->execute();
  const auto projection =
      std::make_shared<opossum::Projection>(table_scan, expression_vector(a_b, a_a, add_(a_b, a_a)));
  projection->execute();

  const auto input_segment = dict_table->get_chunk(ChunkID{0})->get_segment(ColumnID{0});
  const auto input_dictionary_segment = std::dynamic_pointer_cast<DictionarySegment<int>>(input_segment);
  ASSERT_TRUE(input_dictionary_segment);

  const auto& output_table = projection->get_output();
  ASSERT_EQ(output_table->chunk_count(), 3);

  // Note that column a now has column id b
  const auto output_segment = output_table->get_chunk(ChunkID{0})->get_segment(ColumnID{1});
  const auto output_dictionary_segment = std::dynamic_pointer_cast<DictionarySegment<int>>(output_segment);
  ASSERT_TRUE(output_dictionary_segment);

  EXPECT_TRUE(output_dictionary_segment->dictionary() == input_dictionary_segment->dictionary());

  const auto& output_attribute_vector = output_dictionary_segment->attribute_vector();
  EXPECT_EQ(output_attribute_vector->size(), 1);

  const auto output_attribute_vector_decompressor = output_attribute_vector->create_base_decompressor();
  EXPECT_EQ(output_attribute_vector_decompressor->get(0), ValueID{1});
}
```

Reply:
`// Note that column a now has column id b`: do you mean `column id 1`?